### DISCUSSION TITLE
support private github repositories in homebrew manager

### DISCUSSION THREAD

**Discussion by chrono** [renovatebot/renovate#24867] [Suggest an Idea]
### What would you like Renovate to be able to do?

The homebrew manager should be able to download release archives from a private repository and bump sha256 just like it does with public github repos.

At the moment, when used with a private repository, the manager fails as follows 
```
DEBUG: Failed to download release download for REDACTED - trying archive instead (repository=REDACTED/homebrew-REDACTED, branch=bump-some-formula)
DEBUG: Failed to download archive download for REDACTED - update failed (repository=REDACTED/homebrew-REDACTED, branch=bump-some-formula)
```

The manager does not appear to be using authentication in the course of the download.

https://github.com/renovatebot/renovate/blob/3a58d239ec7467b8b720798a6b2f5d5a74f8fb3c/lib/modules/manager/homebrew/update.ts#L164-L169

### If you have any ideas on how this should be implemented, please tell us here.

Using the already provided github credentials (perhaps like github-releases datasource?), attach auth token to the stream download so that the transfer succeeds.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
How are you running Renovate - self-hosted or hosted app?

**Comment by chrono** [comment]
I'm running renovate self-hosted

**Comment by viceice** [comment]
the `http.stream` call should use our normal hostrules. anything else is a bug. we probably forgot to implement authentication for the stream function, as we didn't need it yet.

**Comment by danieljamesscott** [comment]
I've also bumped into this issue. Anything I can do to help get this fixed.
